### PR TITLE
1 feature 6 dialogue options

### LINKED ISSUE #1: [Feature 6] <Dialogue Options>
The user enters the home screen and starts a new game. When the game starts, PlayerInteractor sets the user (player) starting data in PlayerData. StorylineInteractor grabs the first event of the story. In this event, the narrator gives the player a series of options, each leading to different events. User chooses to take option A, triggering the StorylineInteractor to grab the event corresponding to option A from EventManager. Option A’s event happens to be a CombatEvent. In this event, the player combats an enemy by answering a series of questions collected from QuestionData, each question having its own attack value. Before the player asks the first question, CombatInteractor calls ItemManager to generate an item from ItemData. The item it generates gives the player a permanent HP increase of 5. The player uses the item, prompting PlayerInteractor to raise the player's HP by 5 and remove the item from the player's inventory. Now the player correctly answers their first question, triggering CombatInteractor to reduce the enemy’s HP by 10 stored in EnemyData. The player eventually brings the enemy’s HP to less than or equal to 0, triggering a win condition. This makes the StorylineInteractor grab the next event, not a CombatEvent. The player meets an NPC, which converses with the player and tells the player about different roads the player could take. The player takes path B, triggering StorylineInteractor to grab the next corresponding event. The player saves the game and exits the application.

### PR DESCRIPTION
**Pull request by jshchn4**
StorylineInteractor Pull Request.

### CODE REVIEW

File: src/main/java/StorylineInteractor.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+                }
+            }
+
+        }
+
+        Sound.stopSound(event.getSoundFile);
+
+        //finish the game, dunno final event id
+        if (player.eventID == 1000) {
+            String filename = player.username + ".ser";
```
Comment (alainahu) [inline src/main/java/StorylineInteractor.java]: .username is a private attribute, so I think we need to use the .getUsername() method.

**Comment by yeludan** [comment]
(1) two methods SoundOff and soundOn can be combined into one method:
public void soundSwitch() {
    PlayerAction.getPlayerEventID();
    event = Manager.getEvent(player.eventID);
    if (Sound.getIsPlaying != true) {
        Sound.playSound(event.getSoundFile);
        soundStatus = true;
    }
    else {
        Sound.stopSound();
        soundStatus = false;
    }
}

may add one more instance variable:
int soundStatus = true; //record whether the player want the sound to be on of off. default is on.

(2) if the player want the sound on, play sound. otherwise keep quiet
public static void playEvent(PlayerData player) {
        event = Manager.getEvent(player.eventID);

        if (soundStatus = true) { //if the player want the sound on, play sound. otherwise keep quiet
            Sound.playSound(event.getSoundFile);
        }

        if (event.doesAutosave == True) {
            String filename = player.username + ".ser";
            Save.saveToFile("/savefiles/" + filename, player);
        }

        if (event instanceof CombatEvent) {

            View.display_event(event.getNarration());

            if (Combat.initiateCombat(event) == True) {
                PlayerAction.updateEvent(event.getChoicesNextUUIDs()[0]);
            }
            else {
                ///Player loses
                StorylineInteractor.lose(player);
            }
        }

        else {

            View.display_event(event.getNarration());

            //User input system
            Scanner choice_reader = new
                    Scanner(View.display_event_choices(event.getChoicesNarrations()));

            String choice = choice_reader.nextLine();

            for (String next_event : event.getChoicesNextUUIDs()) {
                if (choice.equals(next_event)) {
                    PlayerAction.updateEvent(choice);
                    break;
                }
            }

        }

        Sound.stopSound();

        //finish the game. final event id is 1000
        if (player.eventID == 1000) {
            String filename = player.getUsername() + ".ser";
            Save.saveToFile("/savefiles/" + filename, player);

            StorylineInteractor.endGame();
        }

        else {
            StorylineInteractor.playEvent(player);
        }
    }
}

(3) loadGame may be changed as follows:
public void loadGame() {
        String username = Login.getCurrentUser();
        String filename = username + ".ser";
        PlayerData player = Load.readFromFile("/savefiles/" + filename); //since it returns a PlayerData
        this.playEvent(player);
    }

(4)
I changed my codes, no need to have getEventNarration or getEventChoices methods, sorry about that!

(5) I think display_event(String narration) takes in a string of narration, not eventID
public static void endGame() {
        //ending text id is 1001
        View.display_event(1001);
        View.returnHomeScreen();
    }

**Comment by yeludan** [comment]
1) soundSwitch() Gwen said that she will be using soundStatus to store user's choice of wanting sound to be on or off. Default is on. Please refer to her.

2) for methods lose() and exitGame(), do we need to add a line to manual save the progress?



3) In StorylineInteractor, we need a method updateEventID(Player player, String narrationChoices), which our StorylineController will call. It takes in a string of narration choices, and update the player eventID with the chosen event UUID.
It's better to have this method in StorylineInteractor is because controller should not be talking to eventManager, it only should talk to StorylineInteractor. While StorylineInteractor can talk to eventManager.


4) In playEvent(PlayerData player)
for (String choice : event.getChoicesNarrations()) {
                choices.append(", ").append(choice);   \\would this be like ",aaaa,bbbb" starting with ","?
            }

**Review by SleepyyPlayz — APPROVED** [review]
looks good

**Pull request merged**